### PR TITLE
Consider KeyboardInterrupt/SystemExit at collection time

### PR DESCRIPTION
**Pull request by anitahammer**
Screenshot from 2024-04-06 21-48-46

### CODE REVIEW

**Review by nicoddemus — CHANGES_REQUESTED** [review]
Thanks @anitahammer indeed this is a much better UX.

Could you please also:

1. Add a test for this to ensure we do not regress in the future. Probably a simple way to simulate this is to make a module raise `KeyboardInterrupt` during import and check that the session was interrupted and the message `!!! KeyboardInterrupt !!!!!!!!!` showed up.
2. Add a CHANGELOG entry
3. Add yourself to `AUTHORS` (optional)

**Comment by anitahammer** [comment]
Thank you for the feedback @nicoddemus ! I've added a test case and verified that it was failing before my changes.

File: changelog/12191.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Finally fixed the issue where keyboard interrupts and system exits were not being respected during the tests collection phase in pytest! No more frustration of tests not ending correctly — happy testing! 🐞🔨🎉
```
Comment (nicoddemus) [inline changelog/12191.bugfix.rst:1]: I like this message, but to keep in tone with the rest of our change log messages:

```suggestion
Keyboard interrupts and system exits are now probably handled during the test collection.
```

**Comment by nicoddemus** [comment]
Thanks @anitahammer for the ping!

**Review by nicoddemus — APPROVED** [review]
Thanks a lot @anitahammer!

**Pull request merged**